### PR TITLE
feat: add integration test for Virtual Machine Objects

### PR DESCRIPTION
**Pull request by leoparente**
image

### CODE REVIEW

File: diode-server/reconciler/changeset/changeset_virt_test.go
Comment (mfiedorowicz) [inline diode-server/reconciler/changeset/changeset_virt_test.go:1]: As we don't create cluster placeholder for virtual machine (if not cluster not ingested alongside VM) anymore, would be advisable to have tests for cases:
- ingest virtualization.virtualmachine with name only (without cluster): this is already covered with latest changes here ✅ 
- ingest virtualization.virtualmachine with name and cluster (both don't exist): to ensure we generate changeset to create both cluster and VM
- ingest virtualization.virtualmachine with name and cluster (cluster exists, VM doesn't): changeset should contain creation of the VM with an existing cluster object
- ingest virtualization.virtualmachine with name and cluster (cluster doesn't exist, VM exists): changeset should contain creation of the cluster and update of the VM with a new cluster object

Does it sound feasible?

**Review by mfiedorowicz — COMMENTED** [review]
Thanks for all of these tests 🙌 Left one question/suggestion.

File: diode-server/reconciler/changeset/changeset_virt_test.go
Comment (leoparente) [reply diode-server/reconciler/changeset/changeset_virt_test.go:1]: I've added the tests, but it raised a question

ingest virtualization.virtualmachine with name and cluster (cluster doesn't exist, VM exists): changeset should contain creation of the cluster and update of the VM with a new cluster object
 - is expected to update for VM be performed?

**Review by mfiedorowicz — APPROVED** [review]
🙌

**Pull request merged**